arangod/Transaction/Methods.cpp

```diff
@@ -633,17 +633,38 @@ transaction::Methods::Methods(
     StorageEngine* engine = EngineSelectorFeature::ENGINE;
 
     _state = engine->createTransactionState(
-       _transactionContextPtr->resolver(), options
+       _transactionContextPtr->vocbase(), options
     ).release();
     TRI_ASSERT(_state != nullptr);
-
+    TRI_ASSERT(_state->isTopLevelTransaction());
+    
     // register the transaction in the context
     _transactionContextPtr->registerTransaction(_state);
   }
 
   TRI_ASSERT(_state != nullptr);
 }
 
+/// @brief create the transaction, used to be UserTransaction
```

Comment:
why the renaming?

Reply:
because the UserTransaction was just a constructor